queenbee/workflow/arguments.py

```diff
@@ -15,7 +15,7 @@ class ArgumentArtifact(BaseModel):
         description='The name of the artifact'
     )
 
-    source: Union[HTTPSource, S3Source, ProjectFolderSource] = Field(
+    source: Union[HTTP, S3, ProjectFolder] = Field(
```

Comment:
Can these three HTTP, S3, ProjectFolder types merged into one type say "ArgumentArtifactSource" and use enum for HTTP, S3, ProjectFolder.

```
class ArgumentArtifactSourceTypes(str, Enum):
    HTTP = 'HTTP'
    S3= 'S3'
    ProjectFolder= 'ProjectFolder'

class ArgumentArtifactSource(_ArtifactSource):
    source_type: ArgumentArtifactSourceTypes
    path: str = Field( ................ )
```

Reply:
I don't think so because each of them are a different object with information and not a string.

Reply:
I agree with @mostaphaRoudsari  on this one. The `HTTP`, `S3` and `ProjectFolder` classes have more properties than just `path` so they are different objects.

```yaml
type: ProjectFolder
path: C:\Users\Pollination\my-model.json
---
type: HTTP
url: https://models.pollination.cloud/my-model.json
---
type: S3
key: path/to/model.json
endpoint: https://s3.pollination.cloud
bucket: my-project
```